consul/src/main/java/com/linecorp/armeria/internal/consul/HealthClient.java

```diff
@@ -67,11 +71,20 @@ private HealthClient(ConsulClient client) {
     /**
      * Returns a healthy endpoint list by service name.
      */
-    CompletableFuture<List<Endpoint>> healthyEndpoints(String serviceName) {
+    CompletableFuture<List<Endpoint>> healthyEndpoints(String serviceName, @Nullable String datacenter,
+                                                       @Nullable String filter) {
         requireNonNull(serviceName, "serviceName");
         final StringBuilder path = new StringBuilder("/health/service/");
         PercentEncoder.encodeComponent(path, serviceName);
-        path.append(PASSING_PARAM);
+        final QueryParamsBuilder paramsBuilder = QueryParams.builder();
+        paramsBuilder.add(PASSING_PARAM, "true");
+        if (datacenter != null) {
+            paramsBuilder.add(DATACENTER_PARAM, datacenter);
+        }
+        if (filter != null) {
+            paramsBuilder.add(FILTER_PARAM, filter);
+        }
+        paramsBuilder.build().appendQueryString(path.append('?'));
```

Comment:
How about extracting this logic to another utility class (e.g. `ConsulClientUtil`)?
so that we can deduplicate the logic?